presto-accumulo/src/main/java/com/facebook/presto/accumulo/AccumuloClient.java

```diff
@@ -0,0 +1,1140 @@
+                            fam = g.getKey();
+                        }
+                    }
+                }
+
+                // randomly generate column family if not found
+                if (fam == null) {
+                    fam = UUID.randomUUID().toString().substring(0, 4);
+                }
+
+                // sanity check for qualifier uniqueness... but, I mean, what are the odds?
+                for (Entry<String, Pair<String, String>> m : mapping.entrySet()) {
+                    if (m.getValue().getRight().equals(qual)) {
+                        tryAgain = true;
+                        break;
+                    }
+                }
+
+                if (!tryAgain) {
+                    mapping.put(column.getName(), Pair.of(fam, qual));
+                }
+            }
+            while (tryAgain);
+        }
+        return mapping;
+    }
+
+    /**
+     * Creates the index tables from the given Accumulo table. No op if
+     * {@link AccumuloTable#isIndexed()} is false.
+     *
+     * @param table Table to create index tables
+     */
+    private void createIndexTables(AccumuloTable table)
+    {
+        try {
+            // If our table is indexed, create the index as well
+            if (table.isIndexed()) {
+                // Create index table and set the locality groups
+                Map<String, Set<Text>> indexGroups = Indexer.getLocalityGroups(table);
+                conn.tableOperations().create(table.getIndexTableName());
+                conn.tableOperations().setLocalityGroups(table.getIndexTableName(), indexGroups);
+
+                // Create index metrics table, attach iterators, and set locality groups
+                conn.tableOperations().create(table.getMetricsTableName());
+                conn.tableOperations().setLocalityGroups(table.getMetricsTableName(), indexGroups);
+                for (IteratorSetting s : Indexer.getMetricIterators(table)) {
+                    conn.tableOperations().attachIterator(table.getMetricsTableName(), s);
+                }
+            }
+        }
+        catch (Exception e) {
+            dropTable(table);
+            throw new PrestoException(INTERNAL_ERROR,
+                    "Accumulo error when creating index tables, check coordinator logs for more detail",
+                    e);
+        }
+    }
+
+    /**
+     * Drops the table metadata from Presto. If this table is internal, the Accumulo
+     * tables are also deleted.
+     *
+     * @param table The Accumulo table
+     */
+    public void dropTable(AccumuloTable table)
+    {
+        SchemaTableName stName = new SchemaTableName(table.getSchema(), table.getTable());
+        String tn = table.getFullTableName();
+
+        // Remove the table metadata from Presto
+        metaManager.deleteTableMetadata(stName);
+
+        // If the table is external, we won't clean up the Accumulo tables
+        if (table.isExternal()) {
```

Comment:
May avoid this when followed https://github.com/prestodb/presto/pull/5030/files#r59885584